### ISSUE TITLE
test_ssh_copy started to fail on ok_(isdir(targetdir))

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#6806]
originally observed/reported in https://github.com/datalad/datalad/pull/6803#issuecomment-1176194792 but then reproduced locally across all branches -- maint, master and even #6799 

<details>
<summary>full dump of all testing attempts</summary> 

```shell
(git)lena:~datalad/datalad-maint[bf-CPs-from-master]git
$> DATALAD_TESTS_SSH=1 python -m nose -s -v  datalad/support/tests/test_sshconnector.py                    
datalad.support.tests.test_sshconnector.test_bundle_invariance ... ok
datalad.support.tests.test_sshconnector.test_ssh_compound_cmds ... ok
datalad.support.tests.test_sshconnector.test_ssh_copy ... FAIL
datalad.support.tests.test_sshconnector.test_ssh_custom_identity_file ... SKIP: Travis-specific '/tmp/dl-test-ssh-id' identity file does not exist
datalad.support.tests.test_sshconnector.test_ssh_git_props ... ok
datalad.support.tests.test_sshconnector.test_ssh_open_close ... ok
datalad.support.tests.test_sshconnector.test_ssh_get_connection ... ok
datalad.support.tests.test_sshconnector.test_ssh_manager_close ... ok
datalad.support.tests.test_sshconnector.test_ssh_manager_close_no_throw ... ok
Versions: annexremote=1.5.0 boto=2.49.0 cmd:7z=16.02 cmd:annex=8.20210223 cmd:bundled-git=UNKNOWN cmd:git=2.35.1 cmd:ssh=9.0p1 cmd:system-git=2.35.1 cmd:system-ssh=9.0p1 datalad=0.16.6+29.g87c0eaa7b exifread=2.3.2 humanize=3.14.0 iso8601=1.0.2 keyring=23.5.0 keyrings.alt=UNKNOWN msgpack=1.0.3 mutagen=1.45.1 platformdirs=2.5.0 requests=2.27.1
Obscure filename: str=b' |;&%b5{}\'"\xce\x94\xd0\x99\xd7\xa7\xd9\x85\xe0\xb9\x97\xe3\x81\x82 .datc ' repr=' |;&%b5{}\'"ΔЙקم๗あ .datc '
Encodings: default='utf-8' filesystem='utf-8' locale.prefered='UTF-8'
Environment: PATH='/home/yoh/proj/datalad/datalad-master/venvs/dev3/bin:/home/yoh/gocode/bin:/home/yoh/gocode/bin:/home/yoh/bin:/home/yoh/.local/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games:/sbin:/usr/sbin:/usr/local/sbin' LANG='en_US.UTF-8' GIT_PAGER='less --no-init --quit-if-one-screen' GIT_CONFIG_PARAMETERS="'init.defaultBranch=dl-test-branch' 'clone.defaultRemoteName=dl-test-remote'" GIT_CONFIG_GLOBAL='/home/yoh/.tmp/datalad_temp_jk9gdig6/.gitconfig' GIT_ASKPASS='true'

======================================================================
FAIL: datalad.support.tests.test_sshconnector.test_ssh_copy
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/yoh/proj/datalad/datalad-master/venvs/dev3/lib/python3.9/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/yoh/proj/datalad/datalad-maint/datalad/tests/utils.py", line 214, in _wrap_skip_if_on_windows
    return func(*args, **kwargs)
  File "/home/yoh/proj/datalad/datalad-maint/datalad/tests/utils.py", line 288, in _wrap_skip_ssh
    return func(*args, **kwargs)
  File "/home/yoh/proj/datalad/datalad-maint/datalad/tests/utils.py", line 874, in _wrap_with_tempfile
    return t(*(arg + (filename,)), **kw)
  File "/home/yoh/proj/datalad/datalad-maint/datalad/tests/utils.py", line 874, in _wrap_with_tempfile
    return t(*(arg + (filename,)), **kw)
  File "/home/yoh/proj/datalad/datalad-maint/datalad/tests/utils.py", line 874, in _wrap_with_tempfile
    return t(*(arg + (filename,)), **kw)
  File "/home/yoh/proj/datalad/datalad-maint/datalad/support/tests/test_sshconnector.py", line 245, in test_ssh_copy
    ok_(isdir(targetdir))
AssertionError: None

----------------------------------------------------------------------
Ran 9 tests in 1.045s

FAILED (SKIP=1, failures=1)
(dev3) 1 35144 ->1.....................................:Wed 06 Jul 2022 10:44:08 AM EDT:.
(git)lena:~datalad/datalad-maint[bf-CPs-from-master]git
$> git cherry-pick 4ffb0010a8b248c378f9adb6265b6a31835afdd3                           
(dev3) 1 35144 ->130.....................................:Wed 06 Jul 2022 10:45:58 AM EDT:.
(git)lena:~datalad/datalad-maint[bf-CPs-from-master]git
$> gitk --all
(dev3) 1 35145.....................................:Wed 06 Jul 2022 10:46:45 AM EDT:.
(git)lena:~datalad/datalad-maint[bf-CPs-from-master]git
$> git co 6c67ee806943eab6082a1a250bb5e71dadf39413
Note: switching to '6c67ee806943eab6082a1a250bb5e71dadf39413'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at 6c67ee806 Merge 4ffb0010a8b248c378f9adb6265b6a31835afdd3 into dd6e7952e814f2bd37256a0e8cafa6cc6ec8dc0a
(dev3) 1 35146.....................................:Wed 06 Jul 2022 10:46:50 AM EDT:.
(git)lena:~datalad/datalad-maint[pull/origin/6799/merge]git
$> git show
commit 6c67ee806943eab6082a1a250bb5e71dadf39413 (HEAD, refs/pull/origin/6799/merge)
Merge: dd6e7952e 4ffb0010a
Author: Christian Mönch <christian.moench@web.de>
Date:   Wed Jul 6 12:18:57 2022 +0200

    Merge 4ffb0010a8b248c378f9adb6265b6a31835afdd3 into dd6e7952e814f2bd37256a0e8cafa6cc6ec8dc0a

(dev3) 1 35147.....................................:Wed 06 Jul 2022 10:46:51 AM EDT:.
(git)lena:~datalad/datalad-maint[pull/origin/6799/merge]git
$> DATALAD_TESTS_SSH=1 python -m pytest -s -v datalad/support/tests/test_sshconnector.py
====================================================== test session starts ======================================================
platform linux -- Python 3.9.12, pytest-7.1.2, pluggy-1.0.0 -- /home/yoh/proj/datalad/datalad-master/venvs/dev3/bin/python
cachedir: .pytest_cache
rootdir: /home/yoh/proj/datalad/datalad-maint, configfile: tox.ini
plugins: cov-3.0.0, fail-slow-0.2.0, memray-1.0.0, forked-1.4.0, xdist-2.5.0
collected 9 items                                                                                                               

datalad/support/tests/test_sshconnector.py::test_ssh_get_connection PASSED
datalad/support/tests/test_sshconnector.py::test_ssh_open_close PASSED
datalad/support/tests/test_sshconnector.py::test_ssh_manager_close PASSED
datalad/support/tests/test_sshconnector.py::test_ssh_manager_close_no_throw PASSED
datalad/support/tests/test_sshconnector.py::test_ssh_copy FAILED
datalad/support/tests/test_sshconnector.py::test_ssh_compound_cmds PASSED
datalad/support/tests/test_sshconnector.py::test_ssh_custom_identity_file SKIPPED (Travis-specific '/tmp/dl-test-ssh-...)
datalad/support/tests/test_sshconnector.py::test_ssh_git_props PASSED
datalad/support/tests/test_sshconnector.py::test_bundle_invariance PASSEDVersions: annexremote=1.5.0 boto=2.49.0 cmd:7z=16.02 cmd:annex=8.20210223 cmd:bundled-git=UNKNOWN cmd:git=2.35.1 cmd:ssh=9.0p1 cmd:system-git=2.35.1 cmd:system-ssh=9.0p1 datalad=0.16.6+252.g6c67ee806 exifread=2.3.2 humanize=3.14.0 iso8601=1.0.2 keyring=23.5.0 keyrings.alt=UNKNOWN msgpack=1.0.3 mutagen=1.45.1 platformdirs=2.5.0 requests=2.27.1
Obscure filename: str=b' |;&%b5{}\'"\xce\x94\xd0\x99\xd7\xa7\xd9\x85\xe0\xb9\x97\xe3\x81\x82 .datc ' repr=' |;&%b5{}\'"ΔЙקم๗あ .datc '
Encodings: default='utf-8' filesystem='utf-8' locale.prefered='UTF-8'
Environment: PATH='/home/yoh/proj/datalad/datalad-master/venvs/dev3/bin:/home/yoh/gocode/bin:/home/yoh/gocode/bin:/home/yoh/bin:/home/yoh/.local/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games:/sbin:/usr/sbin:/usr/local/sbin' LANG='en_US.UTF-8' GIT_PAGER='less --no-init --quit-if-one-screen' GIT_CONFIG_PARAMETERS="'init.defaultBranch=dl-test-branch' 'clone.defaultRemoteName=dl-test-remote'" GIT_CONFIG_GLOBAL='/home/yoh/.tmp/datalad_temp_bkujcf0d/.gitconfig' GIT_ASKPASS='true'


=========================================================== FAILURES ============================================================
_________________________________________________________ test_ssh_copy _________________________________________________________

sourcedir = '/home/yoh/.tmp/datalad_temp_test_ssh_copyy4w8054w'
sourcefile1 = '/home/yoh/.tmp/datalad_temp_test_ssh_copyamrjsorm'
sourcefile2 = '/home/yoh/.tmp/datalad_temp_test_ssh_copyjv7xfmhx'

    @skip_if_on_windows
    @skip_ssh
    @with_tempfile(mkdir=True)
    @with_tempfile(content='one')
    @with_tempfile(content='two')
    def test_ssh_copy(sourcedir=None, sourcefile1=None, sourcefile2=None):
        port = get_ssh_port('datalad-test')
        remote_url = 'ssh://datalad-test:{}'.format(port)
        manager = SSHManager()
        ssh = manager.get_connection(remote_url)
    
        # copy content of sourcefile3 to an obscurely named file in sourcedir
        obscure_file = get_most_obscure_supported_name()
        obscure_path = opj(sourcedir, obscure_file)
        with open(obscure_path, 'w') as f:
            f.write('three')
    
        # copy first two temp files to remote_url:sourcedir
        sourcefiles = [sourcefile1, sourcefile2]
        ssh.put(sourcefiles, opj(remote_url, sourcedir))
        # copy obscure file to remote_url:sourcedir/'<obscure_file_name>.c opy'
        # we copy to a different name because the test setup maps local dir and
        # remote dir to the same directory on the test machine. That means the file
        # is copied onto itself. With ssh version 9 this leads to an empty file.
        # We perform copy instead of just writing the content to the destination
        # file,  because ww want to ensure that the source file is picked up by
        # 'ssh.put()'.
        ssh.put([obscure_path], opj(remote_url, sourcedir, obscure_file + '.c opy'))
    
        # docs promise that connection is auto-opened in case of multiplex
        if _ssh_manager_is_multiplex:
            ok_(ssh.is_open())
    
        # recursive copy tempdir to remote_url:targetdir
        targetdir = sourcedir + '.c opy'
        ssh.put(sourcedir, opj(remote_url, targetdir),
                recursive=True, preserve_attrs=True)
    
        # check if sourcedir copied to remote_url:targetdir
>       ok_(isdir(targetdir))

datalad/support/tests/test_sshconnector.py:243: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

expr = False, msg = None

    def assert_true(expr, msg=None):
        if msg is None:
>           assert expr
E           assert False

datalad/tests/utils_pytest.py:191: AssertionError
==================================================== short test summary info ====================================================
FAILED datalad/support/tests/test_sshconnector.py::test_ssh_copy - assert False
============================================ 1 failed, 7 passed, 1 skipped in 1.02s =============================================
(dev3) 1 35148 ->1.....................................:Wed 06 Jul 2022 10:46:59 AM EDT:.
(git)lena:~datalad/datalad-maint[pull/origin/6799/merge]git
$> git co master
Previous HEAD position was 6c67ee806 Merge 4ffb0010a8b248c378f9adb6265b6a31835afdd3 into dd6e7952e814f2bd37256a0e8cafa6cc6ec8dc0a
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
(dev3) 1 35149.....................................:Wed 06 Jul 2022 10:47:17 AM EDT:.
(git)lena:~datalad/datalad-maint[master]git
$> DATALAD_TESTS_SSH=1 python -m pytest -s -v datalad/support/tests/test_sshconnector.py
====================================================== test session starts ======================================================
platform linux -- Python 3.9.12, pytest-7.1.2, pluggy-1.0.0 -- /home/yoh/proj/datalad/datalad-master/venvs/dev3/bin/python
cachedir: .pytest_cache
rootdir: /home/yoh/proj/datalad/datalad-maint, configfile: tox.ini
plugins: cov-3.0.0, fail-slow-0.2.0, memray-1.0.0, forked-1.4.0, xdist-2.5.0
collected 9 items                                                                                                               

datalad/support/tests/test_sshconnector.py::test_ssh_get_connection PASSED
datalad/support/tests/test_sshconnector.py::test_ssh_open_close PASSED
datalad/support/tests/test_sshconnector.py::test_ssh_manager_close PASSED
datalad/support/tests/test_sshconnector.py::test_ssh_manager_close_no_throw PASSED
datalad/support/tests/test_sshconnector.py::test_ssh_copy FAILED
datalad/support/tests/test_sshconnector.py::test_ssh_compound_cmds PASSED
datalad/support/tests/test_sshconnector.py::test_ssh_custom_identity_file SKIPPED (Travis-specific '/tmp/dl-test-ssh-...)
datalad/support/tests/test_sshconnector.py::test_ssh_git_props PASSED
datalad/support/tests/test_sshconnector.py::test_bundle_invariance PASSEDVersions: annexremote=1.5.0 boto=2.49.0 cmd:7z=16.02 cmd:annex=8.20210223 cmd:bundled-git=UNKNOWN cmd:git=2.35.1 cmd:ssh=9.0p1 cmd:system-git=2.35.1 cmd:system-ssh=9.0p1 datalad=0.16.6+256.gb52a10f13 exifread=2.3.2 humanize=3.14.0 iso8601=1.0.2 keyring=23.5.0 keyrings.alt=UNKNOWN msgpack=1.0.3 mutagen=1.45.1 platformdirs=2.5.0 requests=2.27.1
Obscure filename: str=b' |;&%b5{}\'"\xce\x94\xd0\x99\xd7\xa7\xd9\x85\xe0\xb9\x97\xe3\x81\x82 .datc ' repr=' |;&%b5{}\'"ΔЙקم๗あ .datc '
Encodings: default='utf-8' filesystem='utf-8' locale.prefered='UTF-8'
Environment: PATH='/home/yoh/proj/datalad/datalad-master/venvs/dev3/bin:/home/yoh/gocode/bin:/home/yoh/gocode/bin:/home/yoh/bin:/home/yoh/.local/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games:/sbin:/usr/sbin:/usr/local/sbin' LANG='en_US.UTF-8' GIT_PAGER='less --no-init --quit-if-one-screen' GIT_CONFIG_PARAMETERS="'init.defaultBranch=dl-test-branch' 'clone.defaultRemoteName=dl-test-remote'" GIT_CONFIG_GLOBAL='/home/yoh/.tmp/datalad_temp_boxz2em5/.gitconfig' GIT_ASKPASS='true'


=========================================================== FAILURES ============================================================
_________________________________________________________ test_ssh_copy _________________________________________________________

sourcedir = '/home/yoh/.tmp/datalad_temp_test_ssh_copyviy4mry_'
sourcefile1 = '/home/yoh/.tmp/datalad_temp_test_ssh_copyeckq02ww'
sourcefile2 = '/home/yoh/.tmp/datalad_temp_test_ssh_copyyexs7037'

    @skip_if_on_windows
    @skip_ssh
    @with_tempfile(mkdir=True)
    @with_tempfile(content="one")
    @with_tempfile(content="two")
    def test_ssh_copy(sourcedir=None, sourcefile1=None, sourcefile2=None):
        port = get_ssh_port('datalad-test')
        remote_url = 'ssh://datalad-test:{}'.format(port)
        manager = SSHManager()
        ssh = manager.get_connection(remote_url)
    
        # write to obscurely named file in sourcedir
        obscure_file = opj(sourcedir, get_most_obscure_supported_name())
        with open(obscure_file, 'w') as f:
            f.write("three")
    
        # copy tempfile list to remote_url:sourcedir
        sourcefiles = [sourcefile1, sourcefile2, obscure_file]
        ssh.put(sourcefiles, opj(remote_url, sourcedir))
        # docs promise that connection is auto-opened in case of multiplex
        if _ssh_manager_is_multiplex:
            ok_(ssh.is_open())
    
        # recursive copy tempdir to remote_url:targetdir
        targetdir = sourcedir + '.c opy'
        ssh.put(sourcedir, opj(remote_url, targetdir),
                recursive=True, preserve_attrs=True)
    
        # check if sourcedir copied to remote_url:targetdir
>       ok_(isdir(targetdir))

datalad/support/tests/test_sshconnector.py:233: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

expr = False, msg = None

    def assert_true(expr, msg=None):
        if msg is None:
>           assert expr
E           assert False

datalad/tests/utils_pytest.py:191: AssertionError
==================================================== short test summary info ====================================================
FAILED datalad/support/tests/test_sshconnector.py::test_ssh_copy - assert False
============================================ 1 failed, 7 passed, 1 skipped in 0.94s =============================================
(dev3) 1 35150 ->1.....................................:Wed 06 Jul 2022 10:47:20 AM EDT:.
(git)lena:~datalad/datalad-maint[master]git
$> DATALAD_TESTS_SSH=1 python -m pytest -s -v datalad/support/tests/test_sshconnector.py
(dev3) 1 35150 ->130.....................................:Wed 06 Jul 2022 10:47:25 AM EDT:.
(git)lena:~datalad/datalad-maint[master]git
$> git co maint
Switched to branch 'maint'
Your branch is up to date with 'origin/maint'.
(dev3) 1 35151.....................................:Wed 06 Jul 2022 10:47:26 AM EDT:.
(git)lena:~datalad/datalad-maint[maint]git
$> DATALAD_TESTS_SSH=1 python -m pytest -s -v datalad/support/tests/test_sshconnector.py
====================================================== test session starts ======================================================
platform linux -- Python 3.9.12, pytest-7.1.2, pluggy-1.0.0 -- /home/yoh/proj/datalad/datalad-master/venvs/dev3/bin/python
cachedir: .pytest_cache
rootdir: /home/yoh/proj/datalad/datalad-maint
plugins: cov-3.0.0, fail-slow-0.2.0, memray-1.0.0, forked-1.4.0, xdist-2.5.0
collected 9 items                                                                                                               

datalad/support/tests/test_sshconnector.py::test_ssh_get_connection PASSED
datalad/support/tests/test_sshconnector.py::test_ssh_open_close ERROR
datalad/support/tests/test_sshconnector.py::test_ssh_manager_close PASSED
datalad/support/tests/test_sshconnector.py::test_ssh_manager_close_no_throw ERROR
datalad/support/tests/test_sshconnector.py::test_ssh_copy ERROR
datalad/support/tests/test_sshconnector.py::test_ssh_compound_cmds PASSED
datalad/support/tests/test_sshconnector.py::test_ssh_custom_identity_file SKIPPED (Travis-specific '/tmp/dl-test-ssh-...)
datalad/support/tests/test_sshconnector.py::test_ssh_git_props PASSED
datalad/support/tests/test_sshconnector.py::test_bundle_invariance ERROR

============================================================ ERRORS =============================================================
_____________________________________________ ERROR at setup of test_ssh_open_close _____________________________________________
file /home/yoh/proj/datalad/datalad-maint/datalad/support/tests/test_sshconnector.py, line 95
  @skip_if_on_windows
  @skip_ssh
  @with_tree(tree={'f0': 'f0', 'f1': 'f1'})
  @with_tempfile(suffix=get_most_obscure_supported_name(),
                 content="1")
  def test_ssh_open_close(tmp_path, tfile1):
E       fixture 'tfile1' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, doctest_namespace, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, testrun_uid, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory, worker_id
>       use 'pytest --fixtures [testpath]' for help on them.

/home/yoh/proj/datalad/datalad-maint/datalad/support/tests/test_sshconnector.py:95
_______________________________________ ERROR at setup of test_ssh_manager_close_no_throw _______________________________________
file /home/yoh/proj/datalad/datalad-maint/datalad/support/tests/test_sshconnector.py, line 180
  @with_tempfile
  def test_ssh_manager_close_no_throw(bogus_socket):
E       fixture 'bogus_socket' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, doctest_namespace, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, testrun_uid, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory, worker_id
>       use 'pytest --fixtures [testpath]' for help on them.

/home/yoh/proj/datalad/datalad-maint/datalad/support/tests/test_sshconnector.py:180
________________________________________________ ERROR at setup of test_ssh_copy ________________________________________________
file /home/yoh/proj/datalad/datalad-maint/datalad/support/tests/test_sshconnector.py, line 206
  @skip_if_on_windows
  @skip_ssh
  @with_tempfile(mkdir=True)
  @with_tempfile(content="one")
  @with_tempfile(content="two")
  def test_ssh_copy(sourcedir, sourcefile1, sourcefile2):
E       fixture 'sourcedir' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, doctest_namespace, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, testrun_uid, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory, worker_id
>       use 'pytest --fixtures [testpath]' for help on them.

/home/yoh/proj/datalad/datalad-maint/datalad/support/tests/test_sshconnector.py:206
___________________________________________ ERROR at setup of test_bundle_invariance ____________________________________________
file /home/yoh/proj/datalad/datalad-maint/datalad/support/tests/test_sshconnector.py, line 308
  @skip_if_on_windows
  @skip_ssh
  @with_tempfile(mkdir=True)
  def test_bundle_invariance(path):
E       fixture 'path' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, doctest_namespace, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, testrun_uid, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory, worker_id
>       use 'pytest --fixtures [testpath]' for help on them.

/home/yoh/proj/datalad/datalad-maint/datalad/support/tests/test_sshconnector.py:308
======================================================= warnings summary ========================================================
../datalad-master/venvs/dev3/lib/python3.9/site-packages/nose/importer.py:12
  /home/yoh/proj/datalad/datalad-master/venvs/dev3/lib/python3.9/site-packages/nose/importer.py:12: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    from imp import find_module, load_module, acquire_lock, release_lock

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
==================================================== short test summary info ====================================================
ERROR datalad/support/tests/test_sshconnector.py::test_ssh_open_close
ERROR datalad/support/tests/test_sshconnector.py::test_ssh_manager_close_no_throw
ERROR datalad/support/tests/test_sshconnector.py::test_ssh_copy
ERROR datalad/support/tests/test_sshconnector.py::test_bundle_invariance
======================================= 4 passed, 1 skipped, 1 warning, 4 errors in 0.73s =======================================
(dev3) 1 35152 ->1.....................................:Wed 06 Jul 2022 10:47:28 AM EDT:.
(git)lena:~datalad/datalad-maint[maint]git
$> DATALAD_TESTS_SSH=1 python -m nose -s -v datalad/support/tests/test_sshconnector.py  
datalad.support.tests.test_sshconnector.test_bundle_invariance ... ok
datalad.support.tests.test_sshconnector.test_ssh_compound_cmds ... ok
datalad.support.tests.test_sshconnector.test_ssh_copy ... FAIL
datalad.support.tests.test_sshconnector.test_ssh_custom_identity_file ... SKIP: Travis-specific '/tmp/dl-test-ssh-id' identity file does not exist
datalad.support.tests.test_sshconnector.test_ssh_git_props ... ok
datalad.support.tests.test_sshconnector.test_ssh_open_close ... ok
datalad.support.tests.test_sshconnector.test_ssh_get_connection ... ok
datalad.support.tests.test_sshconnector.test_ssh_manager_close ... ok
datalad.support.tests.test_sshconnector.test_ssh_manager_close_no_throw ... ok
Versions: annexremote=1.5.0 boto=2.49.0 cmd:7z=16.02 cmd:annex=8.20210223 cmd:bundled-git=UNKNOWN cmd:git=2.35.1 cmd:ssh=9.0p1 cmd:system-git=2.35.1 cmd:system-ssh=9.0p1 datalad=0.16.6+25.g6ec85d4d5 exifread=2.3.2 humanize=3.14.0 iso8601=1.0.2 keyring=23.5.0 keyrings.alt=UNKNOWN msgpack=1.0.3 mutagen=1.45.1 platformdirs=2.5.0 requests=2.27.1
Obscure filename: str=b' |;&%b5{}\'"\xce\x94\xd0\x99\xd7\xa7\xd9\x85\xe0\xb9\x97\xe3\x81\x82 .datc ' repr=' |;&%b5{}\'"ΔЙקم๗あ .datc '
Encodings: default='utf-8' filesystem='utf-8' locale.prefered='UTF-8'
Environment: PATH='/home/yoh/proj/datalad/datalad-master/venvs/dev3/bin:/home/yoh/gocode/bin:/home/yoh/gocode/bin:/home/yoh/bin:/home/yoh/.local/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games:/sbin:/usr/sbin:/usr/local/sbin' LANG='en_US.UTF-8' GIT_PAGER='less --no-init --quit-if-one-screen' GIT_CONFIG_PARAMETERS="'init.defaultBranch=dl-test-branch' 'clone.defaultRemoteName=dl-test-remote'" GIT_CONFIG_GLOBAL='/home/yoh/.tmp/datalad_temp_6krddewq/.gitconfig' GIT_ASKPASS='true'

======================================================================
FAIL: datalad.support.tests.test_sshconnector.test_ssh_copy
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/yoh/proj/datalad/datalad-master/venvs/dev3/lib/python3.9/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/yoh/proj/datalad/datalad-maint/datalad/tests/utils.py", line 214, in _wrap_skip_if_on_windows
    return func(*args, **kwargs)
  File "/home/yoh/proj/datalad/datalad-maint/datalad/tests/utils.py", line 288, in _wrap_skip_ssh
    return func(*args, **kwargs)
  File "/home/yoh/proj/datalad/datalad-maint/datalad/tests/utils.py", line 874, in _wrap_with_tempfile
    return t(*(arg + (filename,)), **kw)
  File "/home/yoh/proj/datalad/datalad-maint/datalad/tests/utils.py", line 874, in _wrap_with_tempfile
    return t(*(arg + (filename,)), **kw)
  File "/home/yoh/proj/datalad/datalad-maint/datalad/tests/utils.py", line 874, in _wrap_with_tempfile
    return t(*(arg + (filename,)), **kw)
  File "/home/yoh/proj/datalad/datalad-maint/datalad/support/tests/test_sshconnector.py", line 235, in test_ssh_copy
    ok_(isdir(targetdir))
AssertionError: None

----------------------------------------------------------------------
Ran 9 tests in 0.499s

FAILED (SKIP=1, failures=1)
```
</details>

local `cmd:ssh=9.0p1` ... downgrading openssh-client somehow rendered ssh server defunct, yet to figure out WTF is going on.

edit: might relate to #6655 - issues which came up with 9.0

**Comment by yarikoptic** [comment]
FTR: downgrade of ssh client and server (locally) via `sudo apt-get install openssh-{client,sftp-server,server}=1:8.4p1-5` does resolve test failure so it is indeed ssh version dependent issue. (thus part of #6655)

**Comment by christian-monch** [comment]
That is related to #6655. I will take a look at it.

**Comment by adswa** [comment]
As far as I can see, this was fixed in #6826. Please reopen if I'm mistaken.
